### PR TITLE
feat: Generate creative advertisements using Gemini and Imagen 3

### PR DESCRIPTION
**Pull request by ankuljain09**
# Description

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [x] You are listed as the author in your notebook or README file.
  - [x] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [x] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: vision/use-cases/advertisment-creation-using-gemini-and-imagen3/GoogleSans-Bold.ttf
Comment (holtskinner) [inline vision/use-cases/advertisment-creation-using-gemini-and-imagen3/GoogleSans-Bold.ttf:1]: Remove this file from the PR

File: vision/use-cases/advertisment-creation-using-gemini-and-imagen3/sample_gemini_imagen_adImage_1.png
Comment (holtskinner) [inline vision/use-cases/advertisment-creation-using-gemini-and-imagen3/sample_gemini_imagen_adImage_1.png:1]: Remove these images from the repo. Add them to Google Cloud Storage. We have a public bucket to use.

File: vision/use-cases/advertisment-creation-using-gemini-and-imagen3/advertisment_creation_using_gemini_and_imagen3.ipynb
Comment (holtskinner) [inline vision/use-cases/advertisment-creation-using-gemini-and-imagen3/advertisment_creation_using_gemini_and_imagen3.ipynb:1]: The folder & notebook name is misspelled. `advertisement`

Comment (holtskinner) [inline vision/use-cases/advertisment-creation-using-gemini-and-imagen3/advertisment_creation_using_gemini_and_imagen3.ipynb:1]: Remove image output from notebook, it makes the file very large and difficult to review.

Comment (holtskinner) [inline vision/use-cases/advertisment-creation-using-gemini-and-imagen3/advertisment_creation_using_gemini_and_imagen3.ipynb:1]: Add typing for all function parameters/return values.

**Review by holtskinner — CHANGES_REQUESTED** [review]

File: vision/use-cases/advertisment-creation-using-gemini-and-imagen3/advertisment_creation_using_gemini_and_imagen3.ipynb
Comment (holtskinner) [reply vision/use-cases/advertisment-creation-using-gemini-and-imagen3/advertisment_creation_using_gemini_and_imagen3.ipynb:1]: Also, I'd recommend just calling it `advertisement-creation` (gemini and imagen3 is redundant)

Comment (ankuljain09) [reply vision/use-cases/advertisment-creation-using-gemini-and-imagen3/advertisment_creation_using_gemini_and_imagen3.ipynb:1]: Removed the image output

File: vision/use-cases/advertisment-creation-using-gemini-and-imagen3/sample_gemini_imagen_adImage_1.png
Comment (ankuljain09) [reply vision/use-cases/advertisment-creation-using-gemini-and-imagen3/sample_gemini_imagen_adImage_1.png:1]: Removed the images from the repo and referred the public GCS paths in the gradio code

File: vision/use-cases/advertisment-creation-using-gemini-and-imagen3/GoogleSans-Bold.ttf
Comment (ankuljain09) [reply vision/use-cases/advertisment-creation-using-gemini-and-imagen3/GoogleSans-Bold.ttf:1]: Removed the file, Added description to help user download the Roboto.ttf file and use it in the code.

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 1]
```diff
@@ -0,0 +1,826 @@
...
...w0pZAyBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQILAbAaHtblzVSoAAAQIECBAgQIAAAQIECBAgQIAAAQIECBAgQIAAgUkCQttJTAoRIECAAAECBAgQIECAAAECBAgQIECAAAECBAgQIEBgNwJC2924qpUAAQIECBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQKTBIS2k5gUIkCAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAwG4EhLa7cVUrAQIECBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQIEJgkIbScxKUSAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAIHdCAhtd+OqVgIECBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQIECEwSENpOYlKIAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAAECuxH46C6qvXv37i6qVScBAgQIECBAgAABAgQIECBAgAABAgQIECBAgAABAgSOWmCTrPTGB2fbtkZ148aNbVWlHgIECBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQIECJy0wNQodqu3R37++edPGk3nCRAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQIECBAgsA2BdbLTra603Ubn1UGAAAECBAgQIECAAAECBAgQIECAAAECBAgQIECAAIHrJLDVlbbXCc5YCRAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQIECBAgsA0Boe02FNVBgAABAgQIECBAgAABAgQIECBAgAABAgQIECBAgACBDQWEthvCOY0AAQIECBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQLbEBDabkNRHQQIECBAgAABAgQIECBAgAABAgQIECBAgAABAgQIENhQQGi7IZzTCBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQIECBAgsA0Boe02FNVBgAABAgQIECBAgAABAgQIECBAgAABAgQIECBAgACBDQWEthvCOY0AAQIECBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQLbEBDabkNRHQQIECBAgAABAgQIECBAgAABAgQIECBAgAABAgQIENhQQGi7IZzTCBAgQIAAAQIECBAgQIAAAQIECBAgQIAAAQIECBAgsA2B/w/qAbY72uQTpwAAAABJRU5ErkJggg==)"
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:88]: Replace the base64 image with this:

```md
![Architecture](https://storage.googleapis.com/github-repo/generative-ai/vision/advertisement-creation/architecture.png)
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 2]
```diff
@@ -0,0 +1,826 @@
...
+        "%pip install --upgrade --user -q google-cloud-aiplatform rembg Pillow opencv-python numpy requests gradio matplotlib"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gsxxTr18LF4g"
+      },
+      "source": [
+        "##Restart runtime\n",
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:127]:
```suggestion
        "## Restart runtime\n",
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 3]
```diff
@@ -0,0 +1,826 @@
...
+        "gemini_flash_model = GenerativeModel(\"gemini-1.5-flash-002\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "XJXc1L8gCHDi"
+      },
+      "source": [
+        "## Generate Images using Imagen3\n",
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:292]:
```suggestion
        "## Generate Images using Imagen 3\n",
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 4]
```diff
@@ -0,0 +1,826 @@
...
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Cymbal: Faster.  Simpler. Stronger.\n"
+          ]
+        }
+      ],
+      "source": [
+        "# Generate ad text using gemini-1.5-flash model\n",
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:388]:
```suggestion
        "# Generate ad text using Gemini 1.5 Flash\n",
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 5]
```diff
@@ -0,0 +1,826 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "v-p4kbGFCMEy"
+      },
+      "source": [
+        "## Add text overlay on the generated images\n",
+        "Before running the below code, download a font file (eg. [Roboto](https://fonts.google.com/selection?selection.family=Roboto:wght@700)) and provide the font .ttf file path in the below variable `downloaded_font_path`"
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:410]:
```suggestion
        "Before running the below code, download a font file (eg. [Roboto](https://fonts.google.com/selection?selection.family=Roboto:wght@700)) and provide the font `.ttf` file path in the below variable `downloaded_font_path`"
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 6]
```diff
@@ -0,0 +1,826 @@
...
+        "    return image.resize((new_width, new_height), Image.BICUBIC)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "KrRMvy8QCwXd"
+      },
+      "source": [
+        "## Styling the Adverstisement Image\n",
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:513]:
```suggestion
        "## Styling the Advertisement Image\n",
```

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -45,6 +45,7 @@
 /generative-ai/speech/use-cases/storytelling                @holtskinner @GoogleCloudPlatform/generative-ai-devrel
 /generative-ai/vision/use-cases                             @iamthuya @GoogleCloudPlatform/generative-ai-devrel
 /generative-ai/vision/getting-started                       @iamthuya @GoogleCloudPlatform/generative-ai-devrel
+/generative-ai/vision/use-cases/advertisment-creation-using-gemini-and-imagen3    @ankuljain09 @koyelguha
```
Comment (holtskinner) [inline .github/CODEOWNERS:48]:
```suggestion
/generative-ai/vision/use-cases/advertisement-creation-using-gemini-and-imagen3    @ankuljain09 @koyelguha
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 8]
```diff
@@ -0,0 +1,826 @@
...
+        "        print()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "78DhEIvffF06"
+      },
+      "source": [
+        "## Generate advertisment text using Gemini"
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:365]:
```suggestion
        "## Generate advertisement text using Gemini"
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb [hunk 9]
```diff
@@ -0,0 +1,826 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 10,
+      "metadata": {
+        "id": "QN6aWwJdX3xY"
+      },
+      "outputs": [],
+      "source": [
+        "# Function to create advertisment\n",
```
Comment (holtskinner) [inline vision/use-cases/advertisement-creation/advertisement_creation.ipynb:552]:
```suggestion
        "# Function to create advertisement\n",
```

File: vision/use-cases/advertisement-creation/advertisement_creation.ipynb (hunk 1 above)
Comment (ankuljain09) [reply vision/use-cases/advertisement-creation/advertisement_creation.ipynb]: Replaced

**Pull request merged**